### ISSUE TITLE
Providing safe cleanup in the OpenSSL command line application

### ISSUE THREAD

**Issue by beldmit** [openssl/openssl#22540]
If we have an intention to make `OPENSSL_cleanup` mandatory in 4.0 (as of https://github.com/openssl/openssl/issues/17469#issuecomment-1780915514) so probably it's worth doing this (moving the command-line openssl utility to be initialized without installing the atexit handler and use the explicit call of `OPENSSL_cleanup`) in the current master branch.

A spin-off to #22508

**Comment by fwh-dc** [comment]
Was this magically done in the atexit() removal process?

**Comment by beldmit** [comment]
Don't know

**Comment by nhorman** [comment]
yes, it was.  The migration guide currently informs users that OPENSSL_cleanup is not called implicitly by the library as an atexit handler any longer, and that users should do so explicitly if they want to clean up memory prior to application exit.

An explicit call to OPENSSL_cleanup was added in the openssl application as part of commit 31659fe32673a6bd66abf3f8a7d803e81c6ffeed

The removal of implicit cleanup generally should not impact users doing memory leak detection using asan or valgrind.  For those use cases in which those tools have been configured to report reachable memory, we have provided a suppression file as part of f4557e314fc041e37b0e6b543ce0d449965ed544 and documented its use in NOTES-VALGRIND.md

**Comment by fwh-dc** [comment]
Thanks for the details, then I think we should close this issue.
